### PR TITLE
Update changelog with unreleased changes.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
Because of a duplicated `0.6.0` section in `CHANGELOG.md` some entries were deleted, then retrieved under a wrong version, this PR aims to fix that.

I'm not sure about #654, as it was added to an already-published `0.6.0` section in https://github.com/handsontable/hyperformula/commit/b5115dcfcdf0d8a9fe1fadeeabc8b0f2e8176b96#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR18, and it looks as if it's still unreleased, is that correct?

### Related issue(s):
1. #675
2. #682

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,27 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+- **Breaking change**: Changed API of many sheet-related methods to take sheetId instead of sheetName as an argument. (#645)
+- **Breaking change**: Removed support for matrix formulas (`{=FORMULA}`) notation. Engine now supports formulas returning array of values (instead of only scalars). (#652)
+- Changed SWITCH function so it takes array as its first argument.
+- **Breaking change**: Removed numeric matrix detection along with matrixDetection and matrixDetectionThreshold config options. (#669)
```
Comment (wojciechczerniak) [inline CHANGELOG.md:13]: switch them, breaking changes together
they are more important than other

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,27 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+### Added
+- Added support for array arithmetic. (#628)
+- Added performance improvements for array handling. (#629)
+- Added ARRAYFORMULA function. (#630)
+- Added FILTER function. (#668)
+- Added ARRAY_CONSTRAIN function. (#661)
+- Added casting to scalars from non-range arrays. (#663)
+- Added support for range interpolation. (#665)
+- Accepting time in JS Date() objects on the input. (#648)
+- Validation of API arguments types for simple types. (#654)
```
Comment (wojciechczerniak) [inline CHANGELOG.md:24]: Can we rewrite those to start with "added" as all other entries?

**Review by wojciechczerniak — APPROVED** [review]
I think everything is here

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,10 +6,30 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+- **Breaking change**: Changed API of many sheet-related methods to take sheetId instead of sheetName as an argument. (#645)
+- **Breaking change**: Removed support for matrix formulas (`{=FORMULA}`) notation. Engine now supports formulas returning array of values (instead of only scalars). (#652)
+- **Breaking change**: Removed numeric matrix detection along with matrixDetection and matrixDetectionThreshold config options. (#669)
+- Changed SWITCH function so it takes array as its first argument.
+
```
Comment (izulin) [inline CHANGELOG.md:14]: Are those really going to be in `unreleased`? Weren't those were already part of the `develop` that went into `master` branch?

Comment (jansiegel) [reply CHANGELOG.md:14]: The last two releases were hotfixes based on `master`.

**Review by izulin — APPROVED** [review]

**Pull request merged**